crates/file_store/src/store.rs

```diff
@@ -11,18 +11,26 @@ use std::{
 
 /// Persists an append-only list of changesets (`C`) to a single file.
 #[derive(Debug)]
-pub struct Store<C>
+pub struct Store<C, V = C>
 where
     C: Sync + Send,
 {
     magic_len: usize,
     db_file: File,
-    marker: PhantomData<C>,
+    marker: PhantomData<(C, V)>,
 }
 
-impl<C> Store<C>
+impl<C, V> Store<C, V>
 where
-    C: Merge
+    C: Into<V>
+        + Merge
+        + Clone
+        + Default
+        + serde::de::DeserializeOwned
+        + core::marker::Send
+        + core::marker::Sync,
+    V: Into<C>
+        + Default
         + serde::Serialize
         + serde::de::DeserializeOwned
         + core::marker::Send
```

Comment:
```suggestion
    C: Into<V> + Merge + Clone + Default,
    V: Into<C>
        + Default
        + serde::Serialize
        + serde::de::DeserializeOwned,
```

There are a lot of unnecessary bounds here.